internal/controllers/utils/types.go

```diff
@@ -31,6 +32,7 @@ var InventoryConditionTypes = struct {
 	DeploymentServerError:     "DeploymentServerError",
 	ResourceServerError:       "ResourceServerError",
 	SmoRegistrationCompleted:  "SmoRegistrationCompleted",
+	DatabaseServerAvailable:   "DatabaseServerAvailable",
```

Comment:
On top of Available should we also have DB inprogres + failed status?

Reply:
I never understood the benefit of having a laundry list of different statuses rather than having just a single state representation in a single condition with an appropriate failure reason... maybe for some things it makes sense but generally I find it awkward.   

Having said that, I can throw in the failed... but the inprogress would be super transient so probably not worth the hassle.

Reply:
Yeah status is a interesting one. Generally it's simply meant for the outside world to read and do something with it...so it's really up to us to decide what'ss useful for them. API convention doc has some good pointers about it https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties (a great doc in general) 

Just Fail should be good enough in this case!